### PR TITLE
Feat/manifest files

### PR DESCRIPTION
**Pull request by Shesekino**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?

include manifestFiles found with globs on the dynamic image scans

### CODE REVIEW

File: lib/types.ts [hunk 1]
```diff
@@ -55,8 +53,25 @@ export interface ScannedProjectCustom {
    * a future change should be mindful but find a way to unify them if possible.
    */
   depTree: DepTree | PkgTree;
-  targetFile?: string;
-  meta?: any;
+  targetFile?: string; // currently used for application-dependencies scans
+  meta?: any; // not to pollute with actual data; reserved for actual metadata
```
Comment (ivanstanev) [inline lib/types.ts:57]: 👍  🙂 👍

File: lib/types.ts [hunk 2]
```diff
@@ -55,8 +53,25 @@ export interface ScannedProjectCustom {
...
+}
+
+enum ScanType {
+  DependencyTree,
+  DependencyGraph,
+  ManifestFiles,
+}
+
+export interface ScannedProjectExtended extends ScannedProjectCustom {
```
Comment (ivanstanev) [inline lib/types.ts:66]: WDYT of naming `ScannedProjectCustom` just `ScannedProject`? And then this new type `ScannedProjectExtended` could become `ScannedProjectCustom` or keep it `ScannedProjectExtended`

File: lib/types.ts [hunk 3]
```diff
@@ -55,8 +53,25 @@ export interface ScannedProjectCustom {
...
+  scanType: ScanType;
+  // unknowingly structured data; determined by `scanType`
+  data: any;
```
Comment (ivanstanev) [inline lib/types.ts:69]: What about typing this as `unknown`? This disallows accessing properties of `data`, unless you assert its type or explicitly cast it (this way the consumer of the plugin response must actively check what they are receiving)

File: lib/types.ts (hunk 2 above)
Comment (Shesekino) [reply lib/types.ts:66]: ```ScannedProjectCustom``` is used to reflect exactly what the CLI wants right now, I'd rather keep it this way
happy to rename ```ScannedProjectExtended``` to whatever makes sense to you

File: lib/types.ts (hunk 3 above)
Comment (Shesekino) [reply lib/types.ts:69]: sounds brilliant I think :)

File: test/system/system.test.ts [hunk 4]
```diff
@@ -357,7 +362,12 @@ test("inspect redis:3.2.11-alpine", (t) => {
     .then((imageId) => {
       expectedImageId = imageId;
       return plugin.inspect(img, dockerFileLocation, {
-        manifestGlobs: ["/etc/redhat-release*", "/etc/foo", "/nonexist/bar"],
+        manifestGlobs: [
+          "/etc/redhat-release*",
```
Comment (mladkau) [inline test/system/system.test.ts:366]: We should also test globs which are not for a specific directory. So the DRA uses for example `"**/package.json"` or the exclusion of `**/node_modules/**`. Unfortunately, when it comes to manifest files, we can't assume in which path we will find them.

File: test/system/system.test.ts [hunk 5]
```diff
@@ -411,6 +421,37 @@ test("inspect redis:3.2.11-alpine", (t) => {
         },
         "deps",
       );
+
+      const manifestFilesResult = res.scannedProjects.find((res) => {
+        if (!("scanType" in res)) {
+          return false;
+        }
+        const scannedProjectExtended = res as ScannedProjectExtended;
+        if (scannedProjectExtended.scanType === ScanType.ManifestFiles) {
```
Comment (mladkau) [inline test/system/system.test.ts:430]: Just a simple return statement is fine here...
```suggestion
        return scannedProjectExtended.scanType === ScanType.ManifestFiles
```

File: test/system/system.test.ts (hunk 4 above)
Comment (mladkau) [reply test/system/system.test.ts:366]: The current list of globs which is used in production can be found here: https://github.com/snyk/docker-registry-agent/blob/develop/src/routes/discover.ts#L160

File: lib/response-builder.ts [hunk 6]
```diff
@@ -37,6 +37,15 @@ function buildResponse(
     ...applicationDependenciesScanResults,
   ];
 
+  if (manifestFiles.length > 0) {
+    scannedProjects.push({
+      scanType: types.ScanType.ManifestFiles,
+      data: manifestFiles,
+      packageManager: "PLEASE DON'T USE THIS",
```
Comment (shaimendel) [inline lib/response-builder.ts:44]: not sure what I think about that :)

File: test/system/system.test.ts [hunk 7]
```diff
@@ -357,7 +362,13 @@ test("inspect redis:3.2.11-alpine", (t) => {
     .then((imageId) => {
       expectedImageId = imageId;
       return plugin.inspect(img, dockerFileLocation, {
-        manifestGlobs: ["/etc/redhat-release*", "/etc/foo", "/nonexist/bar"],
+        manifestGlobs: [
```
Comment (shaimendel) [inline test/system/system.test.ts:365]: I think there are also globs for directories to ignore right?

**Pull request merged**